M1FExplained/Chapter10/Exercises/Exercise05.lean

```diff
@@ -26,31 +35,150 @@ lemma part_a (m n a : ℤ) (hmn : Int.gcd m n = 1) (hm : m ∣ a) (hn : n ∣ a)
     exact h₂ 
   match hc with 
   |⟨e, he⟩ => 
+  -- e is the number such that n * e = k₁. It also is the case that (m * n) * e = a, proving our statment. 
   use e
   calc
     a = k₁ * m := by exact hk₁
     _ = (n * e) * m := by rw [he]
     _ = m * n * e := by ring
 
-lemma part_b (m n : ℤ) (hmn : Int.gcd m n ≠ 1) : ∃ (a : ℤ), (m ∣ a) ∧ (n ∣ a) ∧ (¬(m * n) ∣ a) := by sorry
-  
+-- part b
+
+/- 
+Show that the conclusion of part a is false if m and n are not coprime (i.e, show that if  and n
+are not coprime, here exists an integer a such that m ∣ a and n ∣ a but ¬ m * n ∣ a)
+-/
+
+/- 
+Note that this statment of the question as in the book is incorrect. If we restrict m and n to be 
+non-zero, then the statment is true, so that is what we will prove.
+
+To help, we first define two helper lemmas as the solution is longwinded.
+-/
+
+-- helper_1 is a simple inequality involving absolute values.
+
+lemma helper_1 (m n M : ℤ) (hM : 1 < abs M) (hM' : M ∣ m * n) (hm : m ≠ 0) (hn : n ≠ 0) : abs (m * n / M)  < abs (m * n) := by
+match hM' with 
+|⟨k, hk⟩ => 
+have hkn0 : 0 < abs k := by {
+  rw [abs_pos]
+  intro h
+  rw [h] at hk
```

Comment:
```suggestion
  rintro rfl
```
`rintro rfl` is a nice trick; it means "*define* k to be 0" as opposed to "let h be a proof that k is 0". It means that you don't have to do the rewrite because all `k`s turn into 0s automatically.